test/ngResource/resourceSpec.js

```diff
@@ -2101,6 +2101,28 @@ describe('cancelling requests', function() {
 
     expect(creditCard.$cancelRequest).toBe(noop);
   });
+
+  it('should not break when calling old `$cancelRequest` after the response arrives', function() {
+    $httpBackend.whenGET('/CreditCard').respond({});
+
+    var CreditCard = $resource('/CreditCard', {}, {
+      get: {
+        method: 'GET',
+        cancellable: true
+      }
+    });
+
+    var creditCard = CreditCard.get();
+    var cancelRequest = creditCard.$cancelRequest;
+
+    creditCard.$promise.then(function(list) {
+        cancelRequest();
+    });
+
+    $httpBackend.flush();
+
+    expect(creditCard.$cancelRequest).not.toThrow();
```

Comment:
So, I removed lines 2118 - 2121 and added not.toBe(noop) just above 2124, just like you mentioned in your comments. But now the tests are passing before and after the change I made.